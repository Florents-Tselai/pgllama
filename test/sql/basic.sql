/* Create model from file; No errors*/
SELECT '/tmp/phi-2.Q2_K.gguf'::llama_model;

-- ## Model Metadata
SELECT llama_model_desc('/tmp/phi-2.Q2_K.gguf'::llama_model);
SELECT pg_size_pretty(llama_model_size('/tmp/phi-2.Q2_K.gguf'::llama_model));
SELECT llama_model_n_params('/tmp/phi-2.Q2_K.gguf'::llama_model);

-- ## Tokenizing
-- llama-tokenize -m /tmp/phi-2.Q2_K.gguf -p "Hello World" --ids
SELECT llama_tokenize('/tmp/phi-2.Q2_K.gguf', 'Hello World');
SELECT llama_tokenize('/tmp/phi-2.Q2_K.gguf', 'Hello World</s>', add_special => false, parse_special => false);
SELECT llama_tokenize('/tmp/phi-2.Q2_K.gguf', 'Hello World</s>', add_special => true, parse_special => true);
SELECT llama_tokenize('/tmp/phi-2.Q2_K.gguf', '', add_special => true, parse_special => true);

