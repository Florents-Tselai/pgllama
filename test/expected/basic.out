/* Create model from file; No errors*/
SELECT '/tmp/phi-2.Q2_K.gguf'::llama_model;
      llama_model      
-----------------------
 phi2 3B Q2_K - Medium
(1 row)

-- ## Model Metadata
SELECT llama_model_desc('/tmp/phi-2.Q2_K.gguf'::llama_model);
   llama_model_desc    
-----------------------
 phi2 3B Q2_K - Medium
(1 row)

SELECT pg_size_pretty(llama_model_size('/tmp/phi-2.Q2_K.gguf'::llama_model));
 pg_size_pretty 
----------------
 1118 MB
(1 row)

SELECT llama_model_n_params('/tmp/phi-2.Q2_K.gguf'::llama_model);
 llama_model_n_params 
----------------------
           2779683840
(1 row)

-- ## Tokenizing
-- llama-tokenize -m /tmp/phi-2.Q2_K.gguf -p "Hello World" --ids
SELECT llama_tokenize('/tmp/phi-2.Q2_K.gguf', 'Hello World');
 llama_tokenize 
----------------
 {15496,2159}
(1 row)

SELECT llama_tokenize('/tmp/phi-2.Q2_K.gguf', 'Hello World</s>', add_special => false, parse_special => false);
     llama_tokenize      
-------------------------
 {15496,2159,3556,82,29}
(1 row)

SELECT llama_tokenize('/tmp/phi-2.Q2_K.gguf', 'Hello World</s>', add_special => true, parse_special => true);
     llama_tokenize      
-------------------------
 {15496,2159,3556,82,29}
(1 row)

SELECT llama_tokenize('/tmp/phi-2.Q2_K.gguf', '', add_special => true, parse_special => true);
 llama_tokenize 
----------------
 {}
(1 row)

